code/game/objects/structures/grille.dm

```diff
@@ -171,28 +171,33 @@
 	if(istype(caller))
 		. = . || (caller.pass_flags & PASSGRILLE)
 
+/obj/structure/grille/wirecutter_act(mob/living/user, obj/item/tool)
+	add_fingerprint(user)
+	if(!shock(user, 100))
+		tool.play_tool_sound(src, 100)
+		deconstruct()
+		return TRUE
+
+/obj/structure/grille/screwdriver_act(mob/living/user, obj/item/tool)
+	add_fingerprint(user)
+	if(!shock(user, 90))
+		tool.play_tool_sound(src, 100)
+		set_anchored(!anchored)
+		user.visible_message(span_notice("[user] [anchored ? "fastens" : "unfastens"] [src]."), \
+			span_notice("You [anchored ? "fasten [src] to" : "unfasten [src] from"] the floor."))
+		return TRUE
+
 /obj/structure/grille/attackby(obj/item/W, mob/user, params)
 	user.changeNext_move(CLICK_CD_MELEE)
 	add_fingerprint(user)
-	if(W.tool_behaviour == TOOL_WIRECUTTER)
-		if(!shock(user, 100))
-			W.play_tool_sound(src, 100)
-			deconstruct()
-	else if((W.tool_behaviour == TOOL_SCREWDRIVER) && (isturf(loc) || anchored))
```

Comment:
I think you lost the anchored check for screwdrivers

Reply:
haha yes I did, good eye